### PR TITLE
METAL-2665 add Prometheus PushGateway as optional config parameter

### PR DESCRIPTION
**Pull request by dabde**
Add functionality to add a prometheus pushgateway env-variable to the cronjob for monitoring backup metrics.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ backup:
     image: kloeckneri/pgdump-gcs:latest
   mysql: {}
 monitoring:
+  # append as an ENV variable "PROMETHEUS_PUSH_GATEWAY" to the backup cronjob
+  promPushGateway: ""
```
Comment (hyunysmile) [inline docs/configuration.md:23]: probably it's not case sensitive and just works for now but what you wrote in https://github.com/kloeckner-i/db-operator/pull/51/files#diff-4bb9f0a337e0474b8eac4826f09f80dc81fc181bd3ef5f311010ff9c123526d8R57 ist `promPushGateWay`

File: pkg/config/test/config_ok.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ backup:
     image: mysqlbackupimage:latest
 monitoring:
   nodeSelector: {}
+  promPushGateway: foo.bar:9200
```
Comment (hyunysmile) [inline pkg/config/test/config_ok.yaml:20]: `promPushGateWay`

File: pkg/config/types.go [hunk 3]
```diff
@@ -51,9 +51,10 @@ type mysqlBackupConfig struct {
 // monitoringConfig defines prometheus exporter configurations
 // which will be created by db-operator when monitoring is enabled
 type monitoringConfig struct {
-	Postgres     postgresMonitoringConfig `yaml:"postgres"`
-	Mysql        mysqlMonitoringConfig    `yaml:"mysql,omitempty"`
-	NodeSelector map[string]string        `yaml:"nodeSelector"`
+	Postgres        postgresMonitoringConfig `yaml:"postgres"`
+	Mysql           mysqlMonitoringConfig    `yaml:"mysql,omitempty"`
+	NodeSelector    map[string]string        `yaml:"nodeSelector"`
+	PromPushGateway string                   `yaml:"promPushGateWay,omitempty"`
```
Comment (hyunysmile) [inline pkg/config/types.go:57]: `yaml:"promPushGateWay,omitempty"`? or `yaml:"promPushGateway,omitempty"`
gateway is a single word.

File: pkg/controller/database/backup/cronjob.go [hunk 4]
```diff
@@ -203,7 +202,14 @@ func postgresEnvVars(dbcr *kciv1alpha1.Database) ([]v1.EnvVar, error) {
 		v1.EnvVar{
 			Name: "GCS_BUCKET", Value: instance.Spec.Backup.Bucket,
 		},
-	}, nil
+	}
+
+	if ok, _ := dbcr.IsMonitoringEnabled(); ok {
```
Comment (hyunysmile) [inline pkg/controller/database/backup/cronjob.go:207]: I think logging error can be helpful :)

File: docs/configuration.md (hunk 1 above)
Comment (dabde) [reply docs/configuration.md:23]: fixed to `promPushGateway` for all cases.

File: pkg/config/test/config_ok.yaml (hunk 2 above)
Comment (dabde) [reply pkg/config/test/config_ok.yaml:20]: fixed to `promPushGateway`

File: pkg/controller/database/backup/cronjob.go (hunk 4 above)
Comment (dabde) [reply pkg/controller/database/backup/cronjob.go]: changed function to use already existing instance pointer.

File: docs/enablingbackup.md [hunk 5]
```diff
@@ -86,3 +86,7 @@ spec:
 The DB Operator will create a kubernetes Cronjob in the same namespace of Database to run backup regularly.
 
 The Cronjob needs permission to push the dump file to the GCS bucket. It will use Secret `google-cloud-storage-bucket-cred`.
+
+## Monitoring
+
+For monitoring a backup job, you can define in the db-operator config a general prometheus pushgateway endpoint (`monitoring.promPushGateway`). If monitoring is enabled, this variable is added to the related backup cronjob enviorment variables as `PROMETHEUS_PUSH_GATEWAY`.
```
[the remark this answers is not in the document]
Comment (dabde) [reply docs/enablingbackup.md]: fixed

File: pkg/config/types.go (hunk 3 above)
Comment (dabde) [reply pkg/config/types.go]: fixed, all used now `promPushGateway`

**Review by hyunysmile — APPROVED** [review]

**Comment by dabde** [comment]
outdated.

**Pull request closed without merging**